pkg/storage/layers/connection.yaml

```diff
@@ -0,0 +1,76 @@
+definitions:
+  Connection:
+    description: Connection information
+    additionalProperties: true
+    properties:
+      db2: 
+        $ref: "#/definitions/db2"
+      s3:
+        $ref: "#/definitions/s3"
+      kafka:
+        $ref: "#/definitions/kafka"
+      fybrik-arrow-flight:
+        $ref: "#/definitions/fybrik-arrow-flight"
+  s3:
+    description: Connection information for S3 compatible object store
+    type: object
+    properties:
+      bucket:
+        type: string
+      endpoint:
+        type: string
+      object_key:
+        type: string
+      region:
+        type: string
+    required:
+    - bucket
+    - endpoint
+    - object_key
+  db2:
+    type: object
+    properties:
+      database:
+        type: string
+      port:
+        type: string
+      ssl:
+        type: string
+      table:
+        type: string
+      url:
+        type: string
+  kafka:
+    type: object
+    properties:
+      bootstrap_servers:
+        type: string
+      key_deserializer:
+        type: string
+      sasl_mechanism:
+        type: string
+      schema_registry:
+        type: string
+      security_protocol:
+        type: string
+      ssl_truststore:
+        type: string
+      ssl_truststore_password:
+        type: string
+      topic_name:
+        type: string
+      value_deserializer:
+        type: string
+  fybrik-arrow-flight:
+    type: object
```

Comment:
description?  Need to mention why this is specific to fybrik and doesn't support arrow flight in general (if that's the case).  If there is nothing fybrik specific, then perhaps this should be renamed?  _If so, probably need to open an issue to address more broadly. _

Reply:
I think the used API is specific to Fybrik, while the arrow-flight in general refers to the data format and not to the protocol. @cdoron - please comment.

Reply:
@shlomitk1 : that is correct. `fybrik-arrow-flight` refers to the specific protocol we defined for fybrik.